### ISSUE TITLE
An auto-enlisted transaction does not log the commit command

### ISSUE THREAD

**Issue by vanbukin** [mysql-net/MySqlConnector#1510]
**Software versions**
MySqlConnector version: 2.3.7
Server type (MySQL, MariaDB, Aurora, etc.) and version: MySQL 8.0.35
.NET version: 8.0.8
ORM NuGet packages and versions: 
Pomelo.EntityFrameworkCore.MySql 8.0.2
Microsoft.EntityFrameworkCore.Relational 8.0.8

**Describe the bug**

I have an extension method that adds DbContext and MySqlDataSource and configures EF Core to work with the MySqlDataSource. 
<details>
  <summary>Extension method</summary>
  
For the specified connection string, the connection string parameters - AutoEnlist and UseXaTransactions are forcibly set to true and false, respectively.
  
  ```csharp
using System;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using MySqlConnector;

namespace MySql.Extensions;

public static class ServiceCollectionExtensions
{
    public static IServiceCollection AddCustomDbContext<TContext>(
        this IServiceCollection services,
        string connectionString,
        string applicationName,
        ServiceLifetime contextLifetime = ServiceLifetime.Scoped,
        ServiceLifetime optionsLifetime = ServiceLifetime.Scoped)
        where TContext : DbContext
    {
        ArgumentNullException.ThrowIfNull(services);
        var connectionStringBuilder = new MySqlConnectionStringBuilder(connectionString)
        {
            AutoEnlist = true,
            UseXaTransactions = false,
            AllowUserVariables = true,
            ApplicationName = applicationName
        };
        var runtimeConnectionString = connectionStringBuilder.ConnectionString;
        var dataSourceKeyProvider = new MySqlDataSourceKeyProvider<TContext>(applicationName);
        services.AddSingleton(dataSourceKeyProvider);
        services.AddKeyedMySqlDataSource(dataSourceKeyProvider.DataSourceKey, runtimeConnectionString);
        services.AddDbContext<TContext>(static (provider, builder) =>
            {
                var dataSourceKeyProvider = provider.GetRequiredService<MySqlDataSourceKeyProvider<TContext>>();
                var dataSourceKey = dataSourceKeyProvider.DataSourceKey;
                var hostEnvironment = provider.GetRequiredService<IHostEnvironment>();
                var mysqlDataSource = provider.GetRequiredKeyedService<MySqlDataSource>(dataSourceKey);
                builder.UseMySql(
                    mysqlDataSource,
                    Constants.MySqlServerVersion,
                    static mysql =>
                    {
                        mysql.MigrationsAssembly(typeof(TContext).Assembly.GetName().FullName);
                        mysql.UseMicrosoftJson();
                    });
                if (hostEnvironment.IsDevelopment())
                {
                    builder.EnableSensitiveDataLogging();
                    builder.EnableDetailedErrors();
                }
            },
            contextLifetime,
            optionsLifetime);
        return services;
    }

    private sealed class MySqlDataSourceKeyProvider<TContext>
        where TContext : DbContext
    {
        public MySqlDataSourceKeyProvider(string dataSourceKey)
        {
            if (string.IsNullOrWhiteSpace(dataSourceKey))
            {
                throw new ArgumentException("Value cannot be null or whitespace.", nameof(dataSourceKey));
            }

            DataSourceKey = dataSourceKey;
        }

        public string DataSourceKey { get; }
    }
}

  ```
  
</details>

I also have a global [action filter](https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-8.0#action-filters) that sets up a transaction for any ASP.NET Core MVC action using `System.Transactions`.
<details>
  <summary>Action filter</summary>
  
```csharp
using System;
using System.Threading.Tasks;
using System.Transactions;
using Microsoft.AspNetCore.Mvc.Filters;

namespace Infrastructure.Mvc.ActionFilters;

public sealed class ExecuteInTransactionActionFilterAttribute : ActionFilterAttribute
{
    public override async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
    {
        ArgumentNullException.ThrowIfNull(context);
        ArgumentNullException.ThrowIfNull(next);
        if (Transaction.Current is null)
        {
            using var transactionScope = new TransactionScope(
                TransactionScopeOption.Required,
                new TransactionOptions
                {
                    IsolationLevel = IsolationLevel.ReadCommitted
                },
                TransactionScopeAsyncFlowOption.Enabled);
            await next();
            transactionScope.Complete();
        }
        else
        {
            await next();
        }
    }
}

  ```
</details>

In the logs, I see all events and commands except for the transaction commit. 
However, the commit itself is executed. 
After debugging the code, I noticed that when [`OnCommit` is called in the `StandardEnlistedTransaction`](https://github.com/mysql-net/MySqlConnector/blob/2.3.7/src/MySqlConnector/Core/StandardEnlistedTransaction.cs#L37-L41) at the `Connection` object, all properties in the `LoggingConfiguration` contain `NullLogger`.

It appears as though the issue lies in this code segment related to closing the connection when using EnlistedTransaction. 
[A new MySqlConnection instance is created, but logging parameters are not passed to it](https://github.com/mysql-net/MySqlConnector/blob/2.3.7/src/MySqlConnector/MySqlConnection.cs#L1177-L1194).
After this, the new `MySqlConnection` instance begins using the global logging configuration, which by default contains `NullLogger`.

**Comment by vanbukin** [comment]
The issue has a workaround in the form of setting global logging parameters. However, your documentation advises against doing this, starting from version 2.3.0.

**Comment by bgrainger** [comment]
Thanks for the clear and detailed bug report!

**Comment by bgrainger** [comment]
Fixed in [2.4.0](https://github.com/mysql-net/MySqlConnector/releases/tag/2.4.0).
